Fusaka upgrade notice page (#1780)
ethereum-optimism/docs: pages/notices/fusaka-notice.mdx

## For chain operators

Update the following:

| Component       | Version                                                                                               | Notes                                                                                                                                                                                                                                                          |
| --------------- | ----------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `proxyd`        | [v4.19.0](https://github.com/ethereum-optimism/infra/releases/tag/proxyd%2Fv4.19.0) or greater        | You need to whitelist `eth_blobBaseFee` rpc if using proxyd for l1 load balancing                                                                                                                                                                              |

> | `op-batcher`    | [v1.16.0](https://github.com/ethereum-optimism/optimism/releases/tag/op-batcher%2Fv1.16.0)            | `OP_BATCHER_TXMGR_ENABLE_CELL_PROOFS`: `true` or equivalent CLI flag needs to be set and the batcher restarted just after Fusaka activates on L1. See [the release notes](https://github.com/ethereum-optimism/optimism/releases)For the `op-batcher` l1 flag. |

Reviewer: ```suggestion
| `op-batcher`    | [v1.16.0](https://github.com/ethereum-optimism/optimism/releases/tag/op-batcher%2Fv1.16.0)            | `OP_BATCHER_TXMGR_ENABLE_CELL_PROOFS`: `true` or equivalent CLI flag needs to be set and the batcher restarted just after Fusaka activates on L1. |
```

---
title: Fusaka upgrade notice
description: Learn how to prepare for the the Fusaka upgrade
lang: en-US
content_type: notice
topic: upgrade-16-changes
personas:
  - chain-operator
categories:
  - security
  - protocol
  - infrastructure
  - interoperability
is_imported_content: 'false'
---

import { Steps, Callout } from 'nextra/components'

# Upgrade 16: Preparing for L1 Fusaka upgrade

This page provides external-facing information about the Fusaka readiness upgrade for OP Stack chains. It summarizes scope, timelines, required actions, and resources for chain operators and partners. This upgrade is necessary for all OP Stack chains deriving from an L1 chain which is due to activate Fusaka (including Ethereum Mainnet and Sepolia).

This is NOT Fusaka adoption on L2. It's a readiness upgrade to make the Optimism protocol compatible with L1 that activates the Fusaka fork. L2 adoption will happen in a future upgrade.

<Callout type="info">

>   All of the tasks below must be done when the OP Stack Fusaka readiness release is available and before Fusaka activates on L1 (October 14th for Sepolia chains).

Reviewer: ```suggestion
  All of the tasks below must be performed before Fusaka activates on L1 (October 14th for Sepolia chains).
```

| `op-geth`       | [v1.101603.1-rc.1](https://github.com/ethereum-optimism/op-geth/releases/tag/v1.101603.1-rc.1)        |                                                                                                                                                                                                                                                                |   |
| `op-challenger` | [v1.6.0-rc.3](https://github.com/ethereum-optimism/optimism/releases/tag/op-challenger%2Fv1.6.0-rc.3) | If deploying to a custom l1, must [supply a custom l1 flag](https://github.com/ethereum-optimism/optimism/releases) Not necessary for chains deriving from Ethereum Mainnet, Sepolia or Hoodi.                                                                 |   |
|                 |                                                                                                       |                                                                                                                                                                                                                                                                |   |

> ### For permissionless fault proof enabled chains

Reviewer: This section is repeated.

  ### Deploy new dispute game contracts

  You will then take the absolute prestate and deploy new `FaultDisputeGame` and `PermissionedDisputeGame` contracts with that value.

  ### Update the DisputeGameFactory

  You will then need to update the `DisputeGameFactory` to point to the new `FaultDisputeGame` and `PermissionedDisputeGame` contracts by calling `DisputeGameFactory.setImplementation`.

  ### Execute the upgrade

  Once your `op-challenger` is ready with the new preimage, you can execute the "Set Dispute Game Implementation" transaction. Please simulate and validate that the expected output prior to executing the transaction.
</Steps>

## For node operators

Update your node software as soon as possible after release:

*   Update `op-node` to v1.14.0-rc.3. If deploying to a custom l1, must supply OP\_NODE\_L1\_GENESIS\_PATH. Not necessary for chains deriving from Ethereum Mainnet, Sepolia or Hoodi.
*   Update `op-geth` to [v1.101603.1-rc.1](https://github.com/ethereum-optimism/op-geth/releases/tag/v1.101603.1-rc.1)

> ### Ensure your l1 beacon node endpoint can serve all blobs

Reviewer: Do you think this applies to chain operators too? As in, chain operators are also node operators. WDYT if we put the whole node operator section above chain operators? I think this reduces the chance chain operators miss this information about beacon endpoints.

---
title: Fusaka upgrade notice
description: Learn how to prepare for the the Fusaka upgrade
lang: en-US
content_type: notice
topic: upgrade-16-changes
personas:
  - chain-operator
categories:
  - security
  - protocol
  - infrastructure
  - interoperability
is_imported_content: 'false'
---

import { Steps, Callout } from 'nextra/components'

> # Upgrade 16: Preparing for L1 Fusaka upgrade

Reviewer: Yes, I prefer to not use numbers at all here.

| `kona-host`     | [v1.1.0-rc.3](https://github.com/op-rs/kona/releases/tag/kona-host%2Fv1.1.0-rc.3)                     |                                                                                                                                                                                                                                                                |
| `kona-client`   | [v1.1.0-rc.3](https://github.com/op-rs/kona/releases/tag/kona-client%2Fv1.1.0-rc.3)                   |                                                                                                                                                                                                                                                                |
| `op-reth`       | [v1.8.2](https://github.com/paradigmxyz/reth/releases/tag/v1.8.2)                                     |                                                                                                                                                                                                                                                                |

### For permissionless fault proof enabled chains

> Chains running permissionless fault proofs will need to deploy new dispute game contracts with new absolute prestates. The new 64 bit version of cannon will be utilized moving forward. The Sepolia Superchain, will utilize [op-program/v1.7.0-rc.2](https://github.com/ethereum-optimism/optimism/tree/op-program/v1.7.0-rc.2). The following permissionless fault proof Sepolia chains are: `Base Sepolia`, `Ink Sepolia`, and `OP Sepolia`.

Reviewer 2: Right, this is done as part of the OPCM upgrade.

---
title: Fusaka upgrade notice
description: Learn how to prepare for the the Fusaka upgrade
lang: en-US
content_type: notice
topic: upgrade-16-changes
personas:
  - chain-operator
categories:
  - security
  - protocol
  - infrastructure
  - interoperability
is_imported_content: 'false'
---

import { Steps, Callout } from 'nextra/components'

# Upgrade 16: Preparing for Fusaka upgrade

This page provides external-facing information about the Fusaka readiness upgrade for OP Stack chains. It summarizes scope, timelines, required actions, and resources for chain operators and partners. This upgrade applies to all OP Stack chains implementing the Fusaka defense changes.

This is NOT Fusaka adoption on L2. It's a readiness upgrade to make the Optimism protocol compatible with L1 that activates the Fusaka fork. L2 adoption will happen in a future upgrade.

<Callout type="info">
  All of the tasks below must be done as soon as the OP Stack Fusaka readiness release is available and before Fusaka activates on L1 (October 14th for Sepolia chains).

> </Callout>

Reviewer: We are aiming to have the releases ready, so don't need to day "as soon as" here.

---
title: Fusaka upgrade notice
description: Learn how to prepare for the the Fusaka upgrade
lang: en-US
content_type: notice
topic: upgrade-16-changes
personas:
  - chain-operator
categories:
  - security
  - protocol
  - infrastructure
  - interoperability
is_imported_content: 'false'
---

import { Steps, Callout } from 'nextra/components'

# Upgrade 16: Preparing for L1 Fusaka upgrade

> 

Reviewer: On https://docs-optimism.netlify.app/notices/pectra-changes we have some "what is included in the L1 fork" material. For Fusaka, we can borrow some material from https://github.com/ethereum-optimism/design-docs/blob/main/security/fma-l1-fusaka.md or better still just link to https://eips.ethereum.org/EIPS/eip-7607.

| ------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
| `proxyd`                                                                                                                                          | [v4.19.0](https://github.com/ethereum-optimism/infra/releases/tag/proxyd%2Fv4.19.0) or greater        | You need to whitelist `eth_blobBaseFee` rpc if using proxyd for l1 load balancing                                                              |
| `op-batcher`                                                                                                                                      | v1.16.0-rc.1                                                                                          |                                                                                                                                                |

> | `op-node`                                                                                                                                         | v1.14.0-rc.3                                                                                          | If deploying to a custom l1, must supply `OP_NODE_L1_GENESIS_PATH`. Not necessary for chains deriving from Ethereum Mainnet, Sepolia or Hoodi. |

Reviewer: I think this env var is incorrect. Can we refer to it in the abstract (i.e. "l1 chain config must be supplied via a new flag"), and then point the reader to the release notes for specifics?

| `op-geth`                                                                                                                                         | [v1.101603.1-rc.1](https://github.com/ethereum-optimism/op-geth/releases/tag/v1.101603.1-rc.1)        |                                                                                                                                                |

> | `op-challenger`/`op-program`                                                                                                                      | [v1.6.0-rc.3](https://github.com/ethereum-optimism/optimism/releases/tag/op-challenger%2Fv1.6.0-rc.3) |                                                                                                                                                |

Reviewer: There will also be a custom l1 flag for op challenger. Again I suggest we point to the release notes for details.

Author: where are the release notes?

Reviewer: They are still in preparation sorry. Could you just put a placeholder for now? You can use https://github.com/ethereum-optimism/optimism/releases for both challenger and op-node.

> | `op-challenger`/`op-program`                                                                                                                      | [v1.6.0-rc.3](https://github.com/ethereum-optimism/optimism/releases/tag/op-challenger%2Fv1.6.0-rc.3) |                                                                                                                                                |

Reviewer 2: In the notes section we should say:

> If deploying to a custom l1, must supply OP_CHALLENGER_L1_GENESIS. Not necessary for chains deriving from Ethereum Mainnet, Sepolia or Hoodi.

> | `OP_BATCHER_TXMGR_ENABLE_CELL_PROOFS`: `true` or equivalent CLI flag needs to be set and the batcher restarted just after Fusaka activates on L1. |                                                                                                       |                                                                                                                                                |

Reviewer 2: Should this be on the op-batcher line in the notes section?

### For permissionless fault proof enabled chains

Chains running permissionless fault proofs will need to:

*   Update `op-challenger` to use the new prestates. The new prestate can be found in [the Superchain Registry](https://github.com/ethereum-optimism/superchain-registry/blob/main/validation/standard/standard-prestates.toml)

> *   Deploy new dispute game contracts with the new prestate

Reviewer 2: These two lines aren't quite right, I would stick with the format from previous releases. U15 has a good one. op-program will be `1.7.0-rc.2` and the absolute prestate will be `0x03f05793ed3abc475137f73b68c80593cbbf40f5d1059d2ad9b8bfcb97fa8929`.

Instead of listing all the chains we could just say something like "chains that are currently in the Superchain Registry".
